src/screens/ChatInfoScreen/ChatInfoScreen.tsx

```diff
@@ -247,6 +249,14 @@ const ChatInfoScreen = () => {
     );
   };
 
+  const getUsername = (item: UserSchema) => {
+    if (item?.anon_user_info_color_code) {
+      return getOfficialAnonUsername(item);
+    }
+
+    return item?.user?.username || item?.username || item?.name;
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `getUsername` function is a new addition and seems to be implemented correctly. It checks if the user has an anonymous color code, and if so, it uses the `getOfficialAnonUsername` function to generate the username. Otherwise, it returns the user's username or name.
```diff
+   const getUsername = (item: UserSchema) => {
+     if (item?.anon_user_info_color_code) {
+       return getOfficialAnonUsername(item);
+     }
+ 
+     return item?.user?.username || item?.username || item?.name;
+   };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->